File: superstitioMain/java/superstitio/cards/general/TempCard/SelfReference.java
package superstitio.cards.general.TempCard;

import com.evacipated.cardcrawl.mod.stslib.fields.cards.AbstractCard.AutoplayField;
import com.megacrit.cardcrawl.actions.common.MakeTempCardInHandAction;
import com.megacrit.cardcrawl.characters.AbstractPlayer;
import com.megacrit.cardcrawl.monsters.AbstractMonster;
import superstitio.DataManager;
import superstitio.cards.general.AbstractTempCard;
import superstitio.delayHpLose.RemoveDelayHpLoseBlock;

public class SelfReference extends AbstractTempCard {
    public static final String ID = DataManager.MakeTextID(SelfReference.class);

    public static final CardType CARD_TYPE = CardType.CURSE;

    public static final CardRarity CARD_RARITY = CardRarity.RARE;

    public static final CardTarget CARD_TARGET = CardTarget.SELF;

    private static final int COST = -2;
    private static final int BLOCK = 5;
    private static final int UPGRADE_PLUS_BLOCK = 3;

    public SelfReference() {
        super(ID, CARD_TYPE, COST, CARD_RARITY, CARD_TARGET);
        this.setupBlock(BLOCK, UPGRADE_PLUS_BLOCK, new RemoveDelayHpLoseBlock());
        AutoplayField.autoplay.set(this, true);
        this.shuffleBackIntoDrawPile = true;
    }

    @Override
    public void use(AbstractPlayer player, AbstractMonster monster) {
        this.dontTriggerOnUseCard = true;
        this.addToBot_gainBlock();
    }

    @Override
    public void triggerOnExhaust() {
        this.addToBot(new MakeTempCardInHandAction(this.makeCopy()));
    }

    @Override
    public void upgradeAuto() {
    }
}

File: superstitioMain/java/superstitio/cards/general/TempCard/SelfReference.kt
package superstitio.cards.general.TempCard

import basemod.AutoAdd
import com.evacipated.cardcrawl.mod.stslib.fields.cards.AbstractCard.AutoplayField
import com.megacrit.cardcrawl.characters.AbstractPlayer
import com.megacrit.cardcrawl.monsters.AbstractMonster
import superstitio.DataManager
import superstitio.SPTT_Color
import superstitio.cards.SetCardColor
import superstitio.cards.general.AbstractTempCard
import superstitio.delayHpLose.RemoveDelayHpLoseBlock
import superstitioapi.utils.ActionUtility

@AutoAdd.Ignore
@SetCardColor(SPTT_Color.TempColor)
class SelfReference : AbstractTempCard(ID, CARD_TYPE, COST, CARD_RARITY, CARD_TARGET)
{
    init
    {
        this.setupBlock(BLOCK, UPGRADE_PLUS_BLOCK, RemoveDelayHpLoseBlock())
        AutoplayField.autoplay[this] = true
        this.dontTriggerOnUseCard = true
        this.shuffleBackIntoDrawPile = true
    }

    override fun use(player: AbstractPlayer?, monster: AbstractMonster?)
    {
        this.dontTriggerOnUseCard = true
        this.addToBot_gainBlock()
    }

    override fun triggerOnExhaust()
    {
        ActionUtility.addToBot_makeTempCardInBattle(SelfReference(), ActionUtility.BattleCardPlace.Hand, upgraded)
    }

    override fun upgradeAuto()
    {
    }

    companion object
    {
        val ID: String = DataManager.MakeTextID(SelfReference::class.java)

        val CARD_TYPE: CardType = CardType.CURSE

        val CARD_RARITY: CardRarity = CardRarity.RARE

        val CARD_TARGET: CardTarget = CardTarget.SELF

        private const val COST = -2
        private const val BLOCK = 5
        private const val UPGRADE_PLUS_BLOCK = 3
    }
}
